src/pages/spawn/spawnHost/fields/UserTagsField.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from "react";
+import styled from "@emotion/styled";
+import { InstanceTag } from "gql/generated/types";
+import { convertArrayToObject } from "utils/array";
+import { UserTagRow } from "./userTagsField/UserTagRow";
+
+export interface UserTagsFieldStateType {
+  addedInstanceTags?: InstanceTag[];
+  deletedInstanceTags?: InstanceTag[];
+}
+
+interface UserTagsFieldProps {
+  data: UserTagsFieldStateType;
+  onChange: React.Dispatch<React.SetStateAction<any>>;
+  instanceTags: InstanceTag[];
+}
+export const UserTagsField: React.FC<UserTagsFieldProps> = ({
+  data,
+  onChange,
+  instanceTags,
+}) => {
+  const userTags = instanceTags?.filter((tag) => tag.canBeModified);
+  const [visibleTags, setVisibleTags] = useState(userTags);
```

Comment:
visibleTags will not be updated on subsequent polled myHosts queries

Reply:
Dont think this is necessarily an issue since this is a user edited field and likely wont be affected by polling since the user has to manually update this field anyways.